dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -15,20 +15,20 @@ class BaseColumnProfileCompiler(with_metaclass(abc.ABCMeta, object)):
     # NOTE: these profilers are ordered. Test functionality if changed.
     _profilers = list()
 
-    def __init__(self, df_series, options=None):
+    def __init__(self, df_series, options=None, data_labeler=None):
```

Comment:
This seems problematic as it forces all compilers, whether they need a datalabeler or not to have that as an input. I think we might want to implement the option only for the datalabeler profilers by building out its init maybe.

Reply:
Isn't that a large, intended aspect built into the library?

Reply:
even we can add additional parameters for the datalabeler profiler, we still need some sort of the self.data_labeler saved in order to run the create_profile() function. If we don't want that self.data_labeler, we have to copy the function create_profile() to overwrite in the datalabeler profiler. I'm not sure there's a better way.

I'd suggest adding a new argument here, called model_associated. So, if the profiler doesn't need a model, it is None. Otherwise, datalabeler profiler needs a model. If later on, we have a ml model for histogram, it will needs a model as well. Thoughts? (I don't really like adding the model into the options. That sounds confusing to me.)

Reply:
Either way we choose to do it, `BaseColumnProfileCompiler` should not need to have an altered input argument as only one compiler requires this functionality. Either we need a new class derived form it, or use an existing class already derived from it.